contracts/router/modules/pool/gmx/GMXV1Module.sol

```diff
@@ -0,0 +1,99 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.17;
+
+import {IERC20, SafeERC20} from "@openzeppelin/contracts-4.5.0/token/ERC20/utils/SafeERC20.sol";
+
+import {IndexedToken, IPoolModule} from "../../../interfaces/IPoolModule.sol";
+
+import {IGMXV1Reader} from "../../../interfaces/gmx/IGMXV1Reader.sol";
+import {IGMXV1Router} from "../../../interfaces/gmx/IGMXV1Router.sol";
+import {IGMXV1Vault} from "../../../interfaces/gmx/IGMXV1Vault.sol";
+
+import {OnlyDelegateCall} from "../OnlyDelegateCall.sol";
+
+/// @notice PoolModule for GMX V1 pools
+/// @dev Implements IPoolModule interface to be used with pools addeded to LinkedPool router
+abstract contract GMXV1Module is OnlyDelegateCall, IPoolModule {
+    using SafeERC20 for IERC20;
+
+    IGMXV1Router public immutable router;
+    IGMXV1Vault public immutable vault;
+    IGMXV1Reader public immutable reader;
+
+    modifier supportedToken(address token) {
+        require(_isSupported(token), "token not supported");
+        _;
+    }
+
+    constructor(address _router, address _reader) {
+        router = IGMXV1Router(_router);
+        vault = IGMXV1Vault(router.vault());
+        reader = IGMXV1Reader(_reader);
+    }
+
+    /// @notice whether token supported by this pool module
+    function _isSupported(address token) internal view virtual returns (bool);
+
+    /// @inheritdoc IPoolModule
+    function poolSwap(
+        address pool,
+        IndexedToken memory tokenFrom,
+        IndexedToken memory tokenTo,
+        uint256 amountIn
+    ) external supportedToken(tokenFrom.token) supportedToken(tokenTo.token) returns (uint256 amountOut) {
+        assertDelegateCall();
+        require(pool == address(vault), "pool != vault");
+        address tokenIn = tokenFrom.token;
+        IERC20(tokenIn).safeApprove(address(router), amountIn);
+
+        address tokenOut = tokenTo.token;
+        uint256 balanceTo = IERC20(tokenOut).balanceOf(address(this));
+
+        address[] memory path = new address[](2);
+        path[0] = tokenFrom.token;
+        path[1] = tokenTo.token;
+        router.swap(path, amountIn, 0, address(this));
+
+        amountOut = IERC20(tokenOut).balanceOf(address(this)) - balanceTo;
+    }
+
+    /// @inheritdoc IPoolModule
+    function getPoolQuote(
+        address pool,
+        IndexedToken memory tokenFrom,
+        IndexedToken memory tokenTo,
+        uint256 amountIn,
+        bool probePaused
+    ) external view supportedToken(tokenFrom.token) supportedToken(tokenTo.token) returns (uint256 amountOut) {
+        require(pool == address(vault), "pool != vault");
+        address tokenIn = tokenFrom.token;
+        address tokenOut = tokenTo.token;
+        (amountOut, ) = reader.getAmountOut(vault, tokenIn, tokenOut, amountIn);
```

Comment:
You need to check `getMaxAmountIn()` here, otherwise quotes for impossible to perform swaps will be returned.

E.g. the current situation (which changes over time):
![image](https://github.com/synapsecns/synapse-contracts/assets/88190723/3044a717-4406-426f-908b-b23e0a07ccc2)